bowling/src/fileReader.cpp

```diff
@@ -0,0 +1,66 @@
+#include <algorithm>
+#include <fstream>
+
+#include "fileReader.hpp"
+
+namespace fs = std::experimental::filesystem;
+
+FileReader::FileReader(const std::string& directory) : directoryPath_(fs::path(directory)), isValid_(false) {
+    
+    if (!checkDirectory()) {
+        return;
+    }
+
+    std::vector<fs::path> files = makeFileList();
+    readFiles(files);
+    isValid_ = true;
+};
+
+bool FileReader::checkDirectory()
+{
+    if (!fs::exists(directoryPath_)) {
+        return false;
+    }
+    if (!fs::is_directory(directoryPath_)) {
+        return false;
+    }
+    return true;
```

Comment:
Might be changed to shorter version: `return fs::exists(directoryPath_) && fs::is_directory(directoryPath_);`

But on the other hand, your version is more straightforward and might be more readable.